api/maven-api-core/src/main/java/org/apache/maven/api/services/xml/XmlWriterRequest.java

```diff
@@ -55,6 +55,13 @@ static <T> XmlWriterRequestBuilder<T> builder() {
         return new XmlWriterRequestBuilder<>();
     }
 
+    default XmlWriterRequest<T> assertWritable() {
+        if (getWriter() == null && getOutputStream() == null && getPath() == null) {
+            throw new IllegalArgumentException("writer, outputStream or path must be non null");
+        }
+        return this;
+    }
```

Comment:
could live somewhere else. Companion or Util.

It belongs here obviously, as everything else require more code, but we can take it away.

Matter of arch-design.